### PR TITLE
[12.0] migration to v12. various modules.

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**

### CODE REVIEW

File: analytic_plan/views/account_analytic_account_view.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+               ref="analytic.view_account_analytic_account_list"/>
+        <field name="arch" type="xml">
+            <field name="balance" position="after">
+                <field name="debit_plan"/>
+                <field name="credit_plan"/>
+                <field name="balance_plan"/>
+                <button string="Planned cost lines"
+                        name="open_plan_cost_tree_view"
+                        type="object"
+                        icon="gtk-open"/>
```
Comment (CasVissers-360ERP) [inline analytic_plan/views/account_analytic_account_view.xml]: Icon is not working and action is missing.

I don't know if it is OK for me to leave feedback here?

Comment (AaronHForgeFlow) [reply analytic_plan/views/account_analytic_account_view.xml]: It is fine @CasVissers Thank you for reviewing. However this PR is kind of a mess. I think it is better to merge this and then fix every module in a single Pull Request

File: .travis.yml [hunk 2]
```diff
@@ -25,19 +25,17 @@ env:
...
+  - git clone --depth=1 https://github.com/Eficent/maintainer-quality-tools.git ${HOME}/maintainer-quality-tools
   - export PATH=${HOME}/maintainer-quality-tools/travis:${PATH}
   - travis_install_nightly
 
 script:
   - travis_run_tests
 
 after_success:
-  - travis_after_tests_success
+  - travis_after_tests_success
```
Comment (MiquelRForgeFlow) [inline .travis.yml:33]: add blank space

File: analytic_journal/model/analytic.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        'account.analytic.journal', 'Analytic Journal')
+
+
+class AccountAnalyticJournal(models.Model):
+    _name = 'account.analytic.journal'
+    _description = 'Analytic Journal'
+
+    name = fields.Char(string='Journal Name', required=True)
+    code = fields.Char(string='Short Code', size=5, required=True)
+    type = fields.Selection([
```
Comment (MiquelRForgeFlow) [inline analytic_journal/model/analytic.py:22]: better if it's renamed to `analytic_journal_type`

File: analytic_plan/wizard/__init__.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+
+# Copyright 2015 Eficent Business and IT Consulting Services S.L.
+# (Jordi Ballester Alomar)
+# Copyright 2016 Matmoz d.o.o.
+# (Matjaž Mozetič)
```
Comment (MiquelRForgeFlow) [inline analytic_plan/wizard/__init__.py:5]: delete copyrights in `__init__.py` files

File: analytic_resource_plan/__init__.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+
+
```
Comment (MiquelRForgeFlow) [inline analytic_resource_plan/__init__.py]: too much blank space here

File: analytic_resource_plan/model/__init__.py [hunk 6]
```diff
@@ -0,0 +1,7 @@
+
+
```
Comment (MiquelRForgeFlow) [inline analytic_resource_plan/model/__init__.py]: too much blank space here

File: analytic_resource_plan/wizard/__init__.py [hunk 7]
```diff
@@ -0,0 +1,4 @@
+
+
```
Comment (MiquelRForgeFlow) [inline analytic_resource_plan/wizard/__init__.py:2]: too much blank space here

File: analytic_resource_plan/wizard/analytic_resource_plan_copy_version.py [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+        source_version = (
+            data.source_version_id if data and
+            data.source_version_id else False
+        )
+        dest_version = (
+            data.dest_version_id if data and
+            data.dest_version_id else False
+        )
+        if dest_version.default_plan:
+            raise Warning(_('''It is prohibited to copy to the default
```
Comment (MiquelRForgeFlow) [inline analytic_resource_plan/wizard/analytic_resource_plan_copy_version.py:63]: UserError

File: analytic_resource_plan/wizard/analytic_resource_plan_copy_version.py [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+                planning version.'''))
+
+        if source_version == dest_version:
+            raise Warning(_('''Choose different source and destination
```
Comment (MiquelRForgeFlow) [inline analytic_resource_plan/wizard/analytic_resource_plan_copy_version.py:67]: UserError

File: project_location/__init__.py [hunk 10]
```diff
@@ -0,0 +1,4 @@
+# © 2017 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from . import model
```
Comment (MiquelRForgeFlow) [inline project_location/__init__.py:4]: add blank space

File: analytic_plan/models/account_analytic_plan_journal.py [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+    code = fields.Char(
+        'Planning Journal Code'
+    )
+    active = fields.Boolean(
+        'Active',
+        help="""If the active field is set to False, it will allow you to
+                hide the analytic journal without removing it.""",
+        default=True
+    )
+    type = fields.Selection(
```
Comment (MiquelRForgeFlow) [inline analytic_plan/models/account_analytic_plan_journal.py:29]: better if it's renamed to `analytic_plan_journal_type`

File: stock_analytic_account/hooks.py [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo.api import Environment, SUPERUSER_ID
+import logging
+logger = logging.getLogger(__name__)
+
+
+def pre_init_hook(cr):
+    env = Environment(cr, SUPERUSER_ID, {})
+    # creating missing anaytic journals if they are not created
+    for aa in env['account.journal'].search(
```
Comment (MiquelRForgeFlow) [inline stock_analytic_account/hooks.py]:
```suggestion
    for aa in env['account.journal'].with_context(active_test=False).search(
```

File: analytic_plan/models/account_analytic_plan.py [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+        self.ensure_one()
+        analytic_journal_obj = self.env['account.analytic.plan.journal']
+
+        prod = False
+        journal = self.journal_id if self.journal_id else analytic_journal_obj
+        if self.product_id:
+            prod = self.product_id
+        if not self.journal_id:
+            j = analytic_journal_obj.search([('type', '=', 'purchase')])
+            journal = j[0] if j and j[0] else False
```
Comment (MiquelRForgeFlow) [inline analytic_plan/models/account_analytic_plan.py]:
```suggestion
            journal = analytic_journal_obj.search([('type', '=', 'purchase')], limit=1)
```

File: analytic_plan/models/account_analytic_plan.py [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
```
Comment (MiquelRForgeFlow) [inline analytic_plan/models/account_analytic_plan.py]:
```suggestion
            journal = analytic_journal_obj.search([('type', '=', 'purchase')], limit=1)
```

File: analytic_plan/models/account_analytic_plan.py [hunk 15]
```diff
@@ -0,0 +1,298 @@
...
+    @api.onchange('journal_id')
+    def journal_change(self):
+        analytic_journal_obj = self.env['account.analytic.plan.journal']
+        prod = False
+        journal = self.journal_id if self.journal_id else self
+        if not self.journal_id:
+            j = analytic_journal_obj.search(
+                [('type', '=', 'purchase')]
+            )
+            journal = j[0] if j and j[0] else False
```
Comment (MiquelRForgeFlow) [inline analytic_plan/models/account_analytic_plan.py]:
```suggestion
            journal = analytic_journal_obj.search([('type', '=', 'purchase')], limit=1)
```

File: project_location/model/project.py [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+    _inherit = 'project.project'
+
+    @api.model
+    def get_parent_stock_data(self):
+        context = self.env.context
+        res = {}
+        if 'default_parent_id' in context and context['default_parent_id']:
+            for project in self.search(
+                    [('analytic_account_id', '=',
+                      context['default_parent_id'])]):
```
Comment (MiquelRForgeFlow) [inline project_location/model/project.py]:
```suggestion
                      context['default_parent_id'])], limit=1):
```

File: .travis.yml [hunk 17]
```diff
@@ -25,19 +25,17 @@ env:
...
-  - TESTS="1" ODOO_REPO="odoo/odoo"
   - TESTS="1" ODOO_REPO="OCA/OCB"
+  - TESTS="1" ODOO_REPO="odoo/odoo"
 
-virtualenv:
-  system_site_packages: true
 
 install:
-  - git clone --depth=1 https://github.com/OCA/maintainer-quality-tools.git ${HOME}/maintainer-quality-tools
+  - git clone --depth=1 https://github.com/Eficent/maintainer-quality-tools.git ${HOME}/maintainer-quality-tools
```
Comment (MiquelRForgeFlow) [inline .travis.yml]:
```suggestion
  - git clone --depth=1 https://github.com/Eficent/maintainer-quality-tools.git -b master-eficent $HOME/maintainer-quality-tools
```

File: .travis.yml [hunk 18]
```diff
@@ -25,19 +25,17 @@ env:
...
   - export PATH=${HOME}/maintainer-quality-tools/travis:${PATH}
```
Comment (MiquelRForgeFlow) [inline .travis.yml]:
```suggestion
   - export PATH=$HOME/maintainer-quality-tools/travis:$PATH
```

File: analytic_journal/model/analytic.py (hunk 3 above)
Comment (AaronHForgeFlow) [reply analytic_journal/model/analytic.py:22]: I agree, but analytic journals are used in so many places that it is not easy to change. Moreover, additional migration script will be needed

**Comment by AaronHForgeFlow** [comment]
Thanks all for your reviews. As long this PR covers too many modules, ad it is green now, it is best to merge it now and work individually with the remaining modules.

If any other issue is discovered, will create separate pull requests for those.

Regards.

**Pull request merged**